app/src/main/java/com/google/droidjump/leveldata/InfiniteLevelData.java

```diff
@@ -17,27 +17,39 @@
 package com.google.droidjump.leveldata;
 
 import static com.google.droidjump.leveldata.JSONKeys.BASE_SPEED_KEY;
+import static com.google.droidjump.leveldata.JSONKeys.GENERATION_FREQUENCIES_KEY;
 import android.content.res.Resources;
 import android.util.Log;
 import org.json.JSONException;
 import org.json.JSONObject;
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * Manages infinite level data.
  */
 public class InfiniteLevelData implements LevelStrategy {
+    private static final ObstacleType[] OBSTACLE_TYPES = ObstacleType.values();
     ObstacleData currentObstacle;
     int baseSpeed;
+    private Map<ObstacleType, Integer> frequencies;
+    private LevelGenerator levelGenerator;
 
     public InfiniteLevelData(int fileId, Resources resources) {
+        frequencies = new HashMap<>();
```

Comment:
nit: it is probably better to create "frequencies" inside "getDataFromFile". This way, it we for some reason want to update the level data, this data will be correctly filled.